File: RequestGenerator/src/main/java/generator/model/Message.java
package generator.model;

import java.util.Objects;

public class Message {
    private final RequestType requestType;
    private final String sourcePath;
    private final String schemaName;
    private final String tableName;
    private final String partitionSpec;

    public Message(RequestType requestType,
                   String sourcePath,
                   String schemaName,
                   String tableName,
                   String partitionSpec) {
        this.requestType = requestType;
        this.sourcePath = sourcePath;
        this.schemaName = schemaName;
        this.tableName = tableName;
        this.partitionSpec = partitionSpec;
    }

    @Override
    public String toString() {
        String result = "";
        String basicMessage = "{\"requestType\":\""
            .concat(requestType.name())
            .concat("\",\n\"sourcePath\":\"")
            .concat(sourcePath)
            .concat("\",\n\"schemaName\":\"")
            .concat(schemaName)
            .concat("\",\n\"tableName\":\"")
            .concat(tableName);
        if ("".equals(partitionSpec)) {
            result = basicMessage.concat("\"}");
        } else {
            result = basicMessage.concat("\",\n\"partitionSpec\":\"")
                .concat(partitionSpec)
                .concat("\"}");
        }
        return result;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Message)) return false;
        Message message = (Message) o;
        return requestType == message.requestType &&
            sourcePath.equals(message.sourcePath) &&
            schemaName.equals(message.schemaName) &&
            tableName.equals(message.tableName) &&
            partitionSpec.equals(message.partitionSpec);
    }

    @Override
    public int hashCode() {
        return Objects.hash(requestType, sourcePath, schemaName, tableName, partitionSpec);
    }
}

File: RequestGenerator/src/main/scala/com/ringcentral/analytics/sns/request/generator/model/Message.scala
package com.ringcentral.analytics.sns.request.generator.model

import com.ringcentral.analytics.sns.request.generator.model.RequestType.RequestType

class Message(requestType: RequestType,
              sourcePath: String,
              schemaName: String,
              tableName: String,
              partitionSpec: String) {

    def getSchemaName(): String = schemaName

    def getTableName(): String = tableName

    def getPartitionSpec(): String = partitionSpec

    override def toString: String = {
        val basicMessage = "{\"requestType\":\"".concat(requestType.toString)
            .concat("\",\n\"sourcePath\":\"")
            .concat(sourcePath)
            .concat("\",\n\"schemaName\":\"")
            .concat(schemaName)
            .concat("\",\n\"tableName\":\"")
            .concat(tableName)
        if ("" == partitionSpec) {
            basicMessage.concat("\"}")
        }
        else {
            basicMessage.concat("\",\n\"partitionSpec\":\"")
                .concat(partitionSpec)
                .concat("\"}")
        }
    }
}
